src/libsyntax/source_map.rs

```diff
@@ -191,6 +191,18 @@ impl SourceMap {
     /// If a file already exists in the source_map with the same id, that file is returned
     /// unmodified
     pub fn new_source_file(&self, filename: FileName, src: String) -> Lrc<SourceFile> {
+        self.try_new_source_file(filename, src)
+            .unwrap_or_else(|OffsetOverflowError| {
+                eprintln!("fatal error: rustc does not support files larger than 4GB");
```

Comment:
Am I overthinking this? Perhaps just `panic!` would be better? Like, it still will be an ICE, but that's not too bad?

Reply:
Better than what?
`FatalError.raise()` is already just panic (except that it's not ICE).
`eprintln` seems useful because otherwise the error message is very generic and it's not clear that the error happens because some file is too large.

Reply:
The alternative I see is to `FatalError.raise()` immediately in `SourceFile::new` and don't thread the error anywhere.

Reply:
>Better than what?

Better than `eprintln!` which feels ... arbitrary? I guess this doesn't really matter anyway, because the probability that someone hits this is negligible. 

>The alternative I see is to FatalError.raise() immediately in SourceFile::new and don't thread the error anywhere.

That doesn't work because libsyntax_pos doesn't know about `error`